src/test/kotlin/com/exactpro/th2/codec/moldudp64/TestMoldUdp64Codec.kt

```diff
@@ -0,0 +1,529 @@
+            val exception = assertThrows<IllegalArgumentException> { codec.encode(parsed) }
+            assertEquals("Negative message count: -1", exception.message)
+        }
+
+        @Test
+        fun `invalid message count`() {
+            val parsed = parsed(header(packetMessageCount = 2U))
+            val exception = assertThrows<IllegalArgumentException> { codec.encode(parsed) }
+            assertEquals("Invalid message count: 2 (expected: 0)", exception.message)
+        }
+
+        @Test
+        fun `big message length`() {
+            val parsed = parsed(
+                header(packetMessageCount = 1U).addField(LENGTHS_FIELD, listOf("65536")),
+                message(byteArrayOf())
+            )
+
+            val exception = assertThrows<IllegalArgumentException> { codec.encode(parsed) }
+            assertEquals("Cannot fit length of message 1 into 2 bytes: 65536", exception.message)
+        }
+
+        @Test
+        fun `negative message length`() {
+            val parsed = parsed(
+                header(packetMessageCount = 1U).addField(LENGTHS_FIELD, listOf("-1")),
+                message(byteArrayOf())
+            )
+
+            val exception = assertThrows<IllegalArgumentException> { codec.encode(parsed) }
+            assertEquals("Negative length of message 1: -1", exception.message)
+        }
+
+        @Test
+        fun `invalid amount of message lengths`() {
+            val parsed = parsed(
+                header(packetMessageCount = 1U, packetMessageLengths = listOf(0U, 1U)),
+                message(byteArrayOf())
+            )
+
+            val exception = assertThrows<IllegalArgumentException> { codec.encode(parsed) }
+            assertEquals("Invalid amount of message lengths: 2 (expected: 1)", exception.message)
+        }
+
+        @Test
+        fun `mismatching message length`() {
+            val parsed = parsed(
+                header(packetMessageCount = 1U, packetMessageLengths = listOf(2U)),
+                message(byteArrayOf(1, 2, 3))
+            )
+
+            val exception = assertThrows<IllegalArgumentException> { codec.encode(parsed) }
+            assertEquals("Mismatching length of message 1: 2 (expected: 3)", exception.message)
+        }
+
+        @Test
+        fun `big actual message length`() {
+            val parsed = parsed(
+                header(packetMessageCount = 1U),
+                message(Random.nextBytes(65536))
+            )
+
+            val exception = assertThrows<IllegalArgumentException> { codec.encode(parsed) }
+            assertEquals("Actual length of message 1 is too big: 65536 (expected at most: 65535)", exception.message)
+        }
+    }
+
+    companion object {
+        private const val SESSION_FIELD = "Session"
+        private const val SEQUENCE_FIELD = "SequenceNumber"
+        private const val COUNT_FIELD = "MessageCount"
+        private const val LENGTHS_FIELD = "MessageLengths"
```

Comment:
I think we should move those fields into a separate object as constants that will be used in the codec as well